File: leetcode/easy/lc169.java
class Solution {
    public int majorityElement(int[] nums) {
        int selectedInt = nums[0];  
        int count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (count == 0) {
                selectedInt = nums[i];
                count = 1;
            } 
            else if (nums[i] == selectedInt) {
                count++;
            } 
            else {
                count--;
            }
        }
        return selectedInt;
    }
}

File: leetcode/easy/python/lc_169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return nums[0]
        threshold = n / 2
        my_map = {nums[0]: 1}
        for i in range(1, n):
            if nums[i] in my_map:
                my_map[nums[i]] += 1
                if my_map[nums[i]] >= threshold:
                    return nums[i]
            else:
                my_map[nums[i]] = 1
        return 0
